tests/Doctrine/Tests/DBAL/Functional/DataAccessTest.php

```diff
@@ -863,6 +863,49 @@ private function skipOci8AndMysqli()
             $this->markTestSkipped('Mysqli driver dont support this feature.');
         }
     }
+
+    public function testIteratorCanBeIteratedMultipleTimes()
+    {
+        $skippedMessage = <<<'MSG'
+Iterators are not rewindable for either PDO or Non PDO drivers. This is because PDO by default is not rewindable and 
+to be consistent we have not implemented rewindable iterators for non PDO drivers. Should PDO ever become rewindable by
+default we should wrap the legacy drivers in Doctrine\DBAL\RewindableGenerator for example on the Mysqli driver you 
+would do something like the following:
+
+return new RewindableGenerator(function() {
+    $this->_stmt->data_seek(0);
+    while ($row = $this->fetch()) {
+        yield $row;
+    }
+});
+
+If this has been done you can include this test. 
+MSG;
+
+        $this->markTestSkipped($skippedMessage);
```

Comment:
@jenkoian I was thinking about mocking the `Connection::prepare()` implementations and returning a `Statement` mock instead of real statement. The statement mock will spy on the method invocations. I don't think we need to make statement classes configurable at least for this test. Not sure how much boilerplate PHPUnit code is required to achieve that.

Do you see any roadblocks with this approach?